### ISSUE TITLE
Do govts need CMSs, not frameworks? (or, "why no Ruby in government?")

### ISSUE THREAD

**Issue by daguar** [codeforamerica/civic-tech-patterns#30]
**TL;DR - I'm summarizing a good recent talk, and think there's a good topic of discussion here: Do governments really just need CMSs? Are we building for our own happiness rather than non-technical user joy and sustainability? (my summary, not her words)**

Sarah Allen ([ultrasaurus](https://twitter.com/ultrasaurus)) -- a returning Presidential Innovation Fellow (PIF) who was at the Smithsonian Institution -- gave a talk at SF Ruby on Monday called:
### "Why no Ruby in government?"

([Sarah's slides are available here.](http://www.slideshare.net/sarah.allen/why-no-ruby-in-gov-pubprf))

Sarah covered a lot about her experiences as a PIF, but also had one big point about the technology/government divide that I think was relevant here:

**In her view, a large majority of government web projects are fundamentally CMS systems with a few customizations specific to the project.**

Thought this would be a good forum to have a discussion of this.

When she got to DC, she didn't _want_ to use PHP, but -- precisely because Ruby doesn't have any well-supported, "clear winner" CMSs -- she couldn't justify rebuilding a lot of the functionality that Wordpress or Drupal give you out-of-the-box.

While there are a bunch of Ruby CMSs ( https://www.ruby-toolbox.com/categories/content_management_systems ) there's nothing at the point of being a clear winner with a strong community providing lots of support.

The fundamental question Sarah posed was: why doesn't Ruby have this? (Ed. note: I think this equally applies to the other in-vogue web languages like Python and Javascript, so it really crosses community boundaries.) Here's a sample of the answers from the crowd:
- The Ruby community focuses on optimizing developer happiness (most new users are already developers); on the other hand, Wordpress in particular was a gateway to programming for largely non-developers. So in Wordpress/Drupal you have communities that care a lot more about non-developer users and deployers.
- Deployment has traditionally been an issue (LAMP stacks are common, cheap, and "dumb"/easy); Heroku's advent has made Ruby deployment easier, but this is pretty recent. (_Ed. note: It can also be harder for government to contract with cloud service providers, since a low monthly credit card charge is so far removed from the normal big procurement processes._)
- Ruby (and Rails in particular) gives you an easy set of tools to build something (MVC/CRUD) while CMSs give you the thing itself, with plug-ins that can be installed by less-technical users pretty easily.

All in all, I think this is a valuable question that technologists working in the civic space should proactively consider. In particular, if there were a "Wordpress for Ruby" (or Python, or Javascript) then it would potentially resolve a lot of the sustainability challenges, given that Wordpress is in a place I'd consider "too big to become defunct."
